frontend/talentsearch/src/js/components/roleSalaryForm/RoleSalaryForm.tsx

```diff
@@ -0,0 +1,351 @@
+/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+import React from "react";
+import { useIntl } from "react-intl";
+import { BasicForm, Checklist } from "@common/components/form";
+import { SubmitHandler } from "react-hook-form";
+import { InformationCircleIcon } from "@heroicons/react/solid";
+import { errorMessages } from "@common/messages";
+import {
+  DialogLevelOne,
+  DialogLevelTwo,
+  DialogLevelThreeLead,
+  DialogLevelThreeAdvisor,
+  DialogLevelFourLead,
+  DialogLevelFourAdvisor,
+} from "./dialogs";
+import { UpdateUserAsUserInput, User } from "../../api/generated";
+import ProfileFormWrapper from "../applicantProfile/ProfileFormWrapper";
+import ProfileFormFooter from "../applicantProfile/ProfileFormFooter";
+
+export type FormValues = Pick<User, "firstName" | "lastName">;
+
+export type RoleSalaryUpdateHandler = (
+  id: string,
+  data: UpdateUserAsUserInput,
+) => void;
+
+export interface RoleSalaryFormProps {
+  initialUser?: User | null;
+  onUpdateRoleSalary?: RoleSalaryUpdateHandler;
+}
+
+export const RoleSalaryForm: React.FunctionComponent<RoleSalaryFormProps> = ({
+  initialUser,
+  onUpdateRoleSalary,
+}) => {
+  const intl = useIntl();
+
+  // modal logic section
+  const [isDialogLevel1Open, setDialogLevel1Open] =
+    React.useState<boolean>(false);
+  const [isDialogLevel2Open, setDialogLevel2Open] =
+    React.useState<boolean>(false);
+  const [isDialogLevel3LeadOpen, setDialogLevel3LeadOpen] =
+    React.useState<boolean>(false);
+  const [isDialogLevel3AdvisorOpen, setDialogLevel3AdvisorOpen] =
+    React.useState<boolean>(false);
+  const [isDialogLevel4ManagerOpen, setDialogLevel4ManagerOpen] =
+    React.useState<boolean>(false);
+  const [isDialogLevel4AdvisorOpen, setDialogLevel4AdvisorOpen] =
+    React.useState<boolean>(false);
+
```

Comment:
I'm guessing you didn't add the `if (!initialUser)` error here since it would fire every time until the logic is in.  Could you copy it in and leave it commented with a TODO so it doesn't get forgotten?